src/player.cpp

```diff
@@ -13332,17 +13341,61 @@ std::string player::weapname(bool charges) const
     }
 }
 
-void player::wield_contents(item *container, bool force_invlet,
-                            const skill_id &/*skill_used*/, int /*volume_factor*/)
+bool player::wield_contents(item *container, int pos, int factor)
 {
-    if(!(container->contents.empty())) {
-        item& weap = container->contents[0];
-        inv.assign_empty_invlet(weap, force_invlet);
-        wield(&(i_add(weap)));
-        container->contents.erase(container->contents.begin());
-    } else {
+    if (container->contents.empty()) {
         debugmsg("Tried to wield contents of empty container (player::wield_contents)");
+        return false;
+    }
+
+    // if index not specified and container has multiple items then ask the player to choose one
+    if (pos < 0) {
+        std::vector<std::string> opts;
+        std::transform(container->contents.begin(), container->contents.end(), std::back_inserter(opts),
+                       [](const item& elem) { return elem.display_name(); });
+
+        if (opts.size() > 1) {
+            pos = (uimenu(false, _("Wield what?"), opts)) - 1;
+        } else {
+            pos = 0;
+        }
+    }
+
+    if (pos > static_cast<int>(container->contents.size())) {
+        debugmsg("Tried to wield non-existent item from container (player::wield_contents)");
```

Comment:
If you're going to check for this error condition, the function should probably left right here. Btw. the check is wrong, it must be `>=`.

You could also combine this check with the above check for an empty container and just check the content size here. If the container is empty (`size() == 0`), no value of `pos` matches. Having both checks seems redundant.